src/cpp/src/utils.cpp

```diff
@@ -408,6 +474,75 @@ void print_compiled_model_properties(ov::CompiledModel& compiled_Model, const ch
         }
     }
 }
+
+std::pair<ov::CompiledModel, KVDesc>
+compile_decoder_for_npu(const std::shared_ptr<ov::Model>& model,
+                        const ov::AnyMap& config,
+                        const KVAxesPosition& kv_pos,
+                        const std::filesystem::path& model_path) {
+    ov::CompiledModel compiled;
+    ov::AnyMap properties = config;
+    KVDesc kv_desc;
+
+    auto blob_path = pop_or_default(properties, "BLOB_PATH", std::string{});
+    const auto export_blob = pop_or_default(properties, "EXPORT_BLOB", false);
+    const bool do_import = (!blob_path.empty() && !export_blob);
+
+    if (do_import) {
+        if (!std::filesystem::exists(blob_path)) {
+            OPENVINO_THROW("Blob file is not found at: " + blob_path);
+        }
+        std::ifstream fin(blob_path, std::ios::in | std::ios::binary);
+        if (!fin.is_open()) {
+            OPENVINO_THROW("Blob file can't be opened: " + blob_path);
+        }
+        compiled = ov::genai::utils::singleton_core().import_model(fin, "NPU", {});
+        kv_desc.max_prompt_len = compiled.get_property("NPUW_MAX_PROMPT_LEN").as<uint32_t>();
+        kv_desc.min_response_len = compiled.get_property("NPUW_MIN_RESPONSE_LEN").as<uint32_t>();
+    } else {
+        kv_desc.max_prompt_len = pop_int_and_cast(properties, "MAX_PROMPT_LEN").value_or(1024u);
+        kv_desc.min_response_len = pop_int_and_cast(properties, "MIN_RESPONSE_LEN").value_or(128u);
+        update_npu_config(properties, model, kv_pos, kv_desc);
+        auto cache_mode = get_option<CacheMode>(config, "CACHE_MODE");
+        // NB: OPTIMIZE_SIZE is only possible when model_path is defined!
+        // Otherwise, switch to OPTIMIZE_SPEED
+        if (cache_mode.has_value() && *cache_mode == CacheMode::OPTIMIZE_SIZE && !model_path.empty()) {
+            compiled = ov::genai::utils::singleton_core().compile_model(model_path, "NPU", properties);
```

Comment:
Thus way we will read the model by `model_path` twice, won't we?

Reply:
I believe your point is the same as @ilya-lavrenov. @smirnov-alexey please explain :)

Reply:
@AsyaPronina  Actually, we read model once in case of `OPTIMIZE_SPEED` and don't explicitly read by default